common/static/common/js/spec/components/paging_collection_spec.js

```diff
@@ -0,0 +1,237 @@
+define(['jquery',
+        'backbone',
+        'underscore',
+        'common/js/components/collections/paging_collection',
+        'common/js/spec_helpers/ajax_helpers',
+        'common/js/spec_helpers/spec_helpers'
+    ],
+    function ($, Backbone, _, PagingCollection, AjaxHelpers, SpecHelpers) {
+        'use strict';
+
+        describe('PagingCollection', function () {
+            var collection,
+                requests,
+                server = {
+                    isZeroIndexed: false,
+                    count: 43,
+                    respond: function () {
+                        var url = requests[requests.length - 1].url;
+                        var page = parseInt(url.match(/page=(\d+)/)[1], 10);
+                        var page_size = parseInt(url.match(/page_size=(\d+)/)[1]);
+                        var page_count = Math.ceil(this.count / page_size);
+
+                        // Make zeroPage consistently start at zero for ease of calculation
+                        var zeroPage = page - (this.isZeroIndexed ? 0 : 1);
+                        if (zeroPage < 0 || zeroPage > page_count) {
+                            AjaxHelpers.respondWithError(requests, 404, {}, requests.length - 1);
+                        } else {
+                            AjaxHelpers.respondWithJson(requests, {
+                                'count': this.count,
+                                'current_page': page,
+                                'num_pages': page_count,
+                                'start': zeroPage * page_size,
+                                'results': []
+                            }, requests.length - 1);
+                        }
+                    }
+                };
+
+            beforeEach(function () {
+                collection = new PagingCollection();
+                collection.perPage = 10;
+                requests = AjaxHelpers.requests(this);
+                server.isZeroIndexed = false;
+                server.count = 43;
+            });
+
+            it('can register sortable fields', function () {
+                collection.registerSortableField('test_field', 'Test Field');
+                expect('test_field' in collection.sortableFields).toBe(true);
+                expect(collection.sortableFields['test_field'].displayName).toBe('Test Field');
+            });
+
+            it('can register filterable fields', function () {
+                collection.registerFilterableField('test_field', 'Test Field');
+                expect('test_field' in collection.filterableFields).toBe(true);
+                expect(collection.filterableFields['test_field'].displayName).toBe('Test Field');
+            });
+
+            it('can set the sort field and get the display name', function () {
+                collection.registerSortableField('test_field', 'Test Field');
+                collection.setSortField('test_field', false);
+                expect(requests.length).toBe(1);
```

Comment:
I think that for most places that you assert that `requests` incremented, you should also assert that whichever request you were interested in passed the expected querystring parameters to the server using `URI`.